packages/features/bookings/Booker/components/BookEventForm/BookEventForm.tsx

```diff
@@ -102,107 +89,87 @@ export const BookEventForm = ({ onCancel }: BookEventFormProps) => {
-    if (Object.keys(formValues).length) return formValues;
-
-    if (!eventType?.bookingFields) {
-      return {};
-    }
-    const querySchema = getBookingResponsesPartialSchema({
-      eventType: {
-        bookingFields: eventType.bookingFields,
-      },
-      view: rescheduleUid ? "reschedule" : "booking",
-    });
-
-    const parsedQuery = querySchema.parse({
-      ...routerQuery,
-      // `guest` because we need to support legacy URL with `guest` query param support
-      // `guests` because the `name` of the corresponding bookingField is `guests`
-      guests: searchParams?.getAll("guests") || searchParams?.getAll("guest"),
-    });
-
-    const defaultUserValues = {
-      email: rescheduleUid
-        ? bookingData?.attendees[0].email
-        : parsedQuery["email"] || session.data?.user?.email || "",
-      name: rescheduleUid
-        ? bookingData?.attendees[0].name
-        : parsedQuery["name"] || session.data?.user?.name || "",
-    };
-
-    if (!isRescheduling) {
-      const defaults = {
-        responses: {} as Partial<z.infer<typeof bookingFormSchema>["responses"]>,
-      };
-
-      const responses = eventType.bookingFields.reduce((responses, field) => {
-        return {
-          ...responses,
-          [field.name]: parsedQuery[field.name] || undefined,
-        };
-      }, {});
-
-      defaults.responses = {
-        ...responses,
-        name: defaultUserValues.name,
-        email: defaultUserValues.email,
-      };
-
-      return defaults;
-    }
-
-    if ((!rescheduleUid && !bookingData) || !bookingData.attendees.length) {
-      return {};
-    }
-    const primaryAttendee = bookingData.attendees[0];
-    if (!primaryAttendee) {
-      return {};
-    }
-
-    const defaults = {
-      responses: {} as Partial<z.infer<typeof bookingFormSchema>["responses"]>,
-    };
+  const { initialValues, key } = useInitialFormValues({
+    eventType,
+    rescheduleUid,
+    isRescheduling,
+  });
 
-    const responses = eventType.bookingFields.reduce((responses, field) => {
-      return {
-        ...responses,
-        [field.name]: bookingData.responses[field.name],
-      };
-    }, {});
-    defaults.responses = {
-      ...responses,
-      name: defaultUserValues.name,
-      email: defaultUserValues.email,
-    };
-    return defaults;
-  }, [eventType?.bookingFields, formValues, isRescheduling, bookingData, rescheduleUid]);
+  return (
+    <BookEventFormChild
+      // initialValues would be null initially as the async schema parsing is happening. Let's show the form in first render without any prefill values
+      // But ensure that when initialValues is available, the form is reset and rerendered with the prefill values
+      key={key}
+      onCancel={onCancel}
+      initialValues={initialValues}
+      isRescheduling={isRescheduling}
+      eventQuery={event}
+      rescheduleUid={rescheduleUid}
+    />
+  );
+};
 
+export const BookEventFormChild = ({
```

Comment:
This separate component allows `BookEventForm` to reset it's state when needed so that RHF can be reinitialized when needed.